apps/system/js/nfc_handover_manager.js

```diff
@@ -14,689 +13,716 @@
-   * @memberof NfcHandoverManager.prototype
-   */
-  _initiateFileTransfer:
-    function _initiateFileTransfer(nfcPeer, blob, requestId) {
-      this._debug('initiateFileTransfer');
-      /*
-       * Initiate a file transfer by sending a Handover Request to the
-       * remote device.
-       */
+      window.addEventListener('bluetooth-disabled', function() {
+        self._debug('bluetooth-disabled');
+        self._clearBluetoothStatus();
+      });
+
+      window.navigator.mozSetMessageHandler('nfc-manager-send-file',
+        function(msg) {
+          self._debug('In New event nfc-manager-send-file' +
+            JSON.stringify(msg));
+          self.handleFileTransfer(msg.peer, msg.blob, msg.requestId);
+      });
+
+      SettingsListener.observe('nfc.debugging.enabled', false,
+                               (enabled) => { this.DEBUG = enabled; });
+
+      window.addEventListener('nfc-transfer-started',
+        this._transferStarted.bind(this)
+      );
+      window.addEventListener('nfc-transfer-completed',
+        this.transferComplete.bind(this)
+      );
+    },
+
+    /**
+     * Save the on/off status of Bluetooth.
+     * @memberof NfcHandoverManager.prototype
+     */
+    _saveBluetoothStatus: function _saveBluetoothStatus() {
+      if (!this.bluetoothStatusSaved) {
+        this.bluetoothStatusSaved = true;
+        this.bluetoothAutoEnabled = !this.bluetooth.enabled;
+      }
+    },
+
+    /**
+     * Restore the Bluetooth status.
+     * @memberof NfcHandoverManager.prototype
+     */
+    _restoreBluetoothStatus: function _restoreBluetoothStatus() {
+      if (!this.isHandoverInProgress() &&
+          BluetoothTransfer.isSendFileQueueEmpty) {
+        if (this.bluetoothAutoEnabled) {
+          this._debug('Disabling Bluetooth');
+          this.settings.createLock().set({'bluetooth.enabled': false});
+          this.bluetoothAutoEnabled = false;
+        }
+        this.bluetoothStatusSaved = false;
+      }
+    },
+
+    /**
+     * Forget a previously saved Bluetooth status.
+     * @memberof NfcHandoverManager.prototype
+     */
+    _clearBluetoothStatus: function _clearBluetoothStatus() {
+      this.bluetoothStatusSaved = false;
+    },
+
+    /*
+     * Performs an action once Bluetooth is enabled. If Bluetooth is disabled,
+     * it is enabled and the action is queued. If Bluetooth is already enabled,
+     * performs the action directly.
+     * @param {Object} action action to be performed
+     * @param {function} action.callback function to be executed
+     * @param {Array} action.args arguments for the function
+     * @memberof NfcHandoverManager.prototype
+     */
+    _doAction: function _doAction(action) {
+      if (!this.bluetooth.enabled) {
+        this._debug('Bluetooth: not yet enabled');
+        this.actionQueue.push(action);
+        if (this.settingsNotified === false) {
+          this.settings.createLock().set({'bluetooth.enabled': true});
```

Comment:
// XXX: Let BluetoothCore do this for you
this.writeSetting({'bluetooth.enabled': false});